Source/Core/Core/GeckoCodeConfig.cpp

```diff
@@ -251,4 +250,21 @@ void SaveCodes(IniFile& inifile, const std::vector<GeckoCode>& gcodes)
   inifile.SetLines("Gecko_Enabled", enabled_lines);
   inifile.SetLines("Gecko_Disabled", disabled_lines);
 }
+
+bool DeserializeLine(const std::string& line, GeckoCode::Code* out)
+{
+  std::vector<std::string> pieces = SplitString(line, ' ');
```

Comment:
I'm not sure if this is ever really a problem in practice, but this function has quite different behavior compared to the old `ss >> std::hex >> new_code.address >> new_code.data;` code. Leading whitespace was previously okay but is now a failure, and any non-space whitespace anywhere near the code parts also no longer works.

If you're okay with breaking behavior anyway I would just go ahead and make this function return `std::optional<GeckoCode::Code>`. The other code above can just construct its own `GeckoCode::Code` with just the `original_line` set in the failure case.

Reply:
Hmm... I hadn't thought about that change in behavior. I think it should be fine for INI files (and having Gecko INI parsing be consistent with AR INI parsing is nice), but I'm unsure if the stuff we download from geckocodes.org might contain odd whitespace. However, we're not actually calling this from the geckocodes.org downloading function, so it should be fine.

Good point about how the code above could handle getting an optional. I've made it do that.